runtime/src/main/java/org/corfudb/runtime/collections/TxnContext.java

```diff
@@ -86,13 +87,15 @@ public TxnContext(@Nonnull final ObjectsView objectsView,
     /**
      * All write api must be validate to ensure that the table belongs to the namespace.
      *
-     * @param table - table being written to
-     * @param <K> -type of the key
-     * @param <V> - type of the payload/value
-     * @param <M> - type of the metadata
+     * @param table         - table being written to
+     * @param key           - key used in the transaction to check for null.
+     * @param validateKey   - should key be validated for null.
+     * @param <K>           - type of the key
+     * @param <V>           - type of the payload/value
+     * @param <M>           - type of the metadata
      */
     private <K extends Message, V extends Message, M extends Message>
-    void validateTableWrittenIsInNamespace(@Nonnull Table<K, V, M> table) {
+    void validateWrite(@Nonnull Table<K, V, M> table, K key, boolean validateKey) {
```

Comment:
Probably should be validateNameSpace and not write. Kind of confusing when invoked for deletes. Also, for writes the Key can never be null, so its not even optional for a map.

Reply:
You'd be surprised then :)
Key can never be null but the way it fails when the table.put() is invoked with the key being null is not pretty..
`ObjectsView 9496 TXEnd[TX[9672]]: Unexpected exception
java.lang.NullPointerException: null
	at org.corfudb.util.serializer.ISerializer.hash(ISerializer.java:123) ~[runtime-3.2.20201109031413.7872.1.jar:?]
	at org.corfudb.runtime.object.transactions.ConflictSetInfo.generateHashFromObject(ConflictSetInfo.java:28) ~[runtime-3.2.20201109031413.7872.1.jar:?]
	at org.corfudb.runtime.object.transactions.ConflictSetInfo.lambda$null$1(ConflictSetInfo.java:41) ~[runtime-3.2.20201109031413.7872.1.jar:?]
	at`
we do need to validate for deletes as well because sometimes buggy code causes callers to send in null into the key.
We need to catch the error much early on for debug-ability.